Automattic/wp-calypso
# Focused Launch: Implement step highlighting logic

#### Changes proposed in this Pull Request

* Implements highlighting logic for each step in the Summary view.

#### Testing instructions

**Important**: Please remember to reset localStorage when trying to clear some pre-selected values when opening and re-openeing the Focus Launch modal.


1. Create a new site via `/start` giving it no title and ensuring it's on the free plan.
2. Go to `/page/[UNLAUNCHED_SITE_CREATED_BY_START]/home?flags=create/focused-launch-flow-calypso`
3. Click "Launch" to open the modal


- [x] Both the Domains Step and the Plans Step should be disabled at this point
- [x] Enter a site title, making sure it's not "Site Title", this should enable the Domains Step
- [x] Change the site title back to "Site Title" to check this disables the Domains Step again
- [x] Making sure the Domains Step is enabled (repeat step 5) no domain (not even the sub-domain) should be pre-selected. The Plans Step should _still_ be disabled at this point.
- [x] Selecting a domain should highlight it as selected, and should now enable the plans step.
- [x] Selecting a plan should highlight it as selected. At this point all steps should be enabled.

Fixes #47458
Fixes #47743
